syserol/regression.py

```diff
@@ -11,53 +11,58 @@
 )
 
 
-def function_elastic_net(function="ADCC"):
+def function_elastic_net():
     """ Predict functions using elastic net according to Alter methods"""
     # Import Luminex, Luminex-IGG, Function, and Glycan into DF
     df = importAlterDF(function=True, subjects=False)
     df_merged = df.dropna()
-    # separate dataframes
-    Y = df_merged[function]
     df_variables = df_merged.drop(["subject"] + functions, axis=1)
+    # separate dataframes
+    function_results = []
+    for function in functions:
+        Y = df_merged[function]
 
-    # perform regression
-    regr = ElasticNetCV(normalize=True, max_iter=10000, cv=30, n_jobs=-1, l1_ratio=0.8)
-    regr.fit(df_variables, Y)
-    enet = ElasticNet(
-        alpha=regr.alpha_, l1_ratio=regr.l1_ratio_, normalize=True, max_iter=10000
-    )
+        # perform regression
+        regr = ElasticNetCV(normalize=True, max_iter=10000, cv=30, n_jobs=-1, l1_ratio=0.8)
+        regr.fit(df_variables, Y)
+        enet = ElasticNet(
+            alpha=regr.alpha_, l1_ratio=regr.l1_ratio_, normalize=True, max_iter=10000
+        )
 
-    Y_pred = cross_val_predict(enet, df_variables, Y, cv=len(Y), n_jobs=-1)
-    return Y, Y_pred, np.sqrt(r2_score(Y, Y_pred))
+        Y_pred = cross_val_predict(enet, df_variables, Y, cv=len(Y), n_jobs=-1)
+        function_results.append((np.sqrt(r2_score(Y, Y_pred)), Y, Y_pred))
+    return function_results
 
 
-def function_prediction(tensorFac, function="ADCC", evaluation="all"):
+def function_prediction(tensorFac, evaluation="all"):
     """ Predict functions using our decomposition and regression methods"""
     func, _ = importFunction()
     indices = AlterIndices()
+    our_function_results = []
+    for function in functions:
+        Y = func[function]
```

Comment:
If you put all the functions as columns in Y, you can predict them all at once (without a loop).

Reply:
Maybe I'm confused, but wouldn't I still need a loop for the rest of the function?

Reply:
No, you can actually do everything here without loops, and that would be ideal for adjusting this later. For example, look at [`r2_score`](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.r2_score.html). If you give it `Y`'s that are matrices, it understands that the rows are samples and the columns are different outputs, then calculates the values for each output.

Reply:
Okay, how do I deal with the fact that for some functions, certain subjects are dropped due to NaN while others aren't?

Reply:
Ergh. Then we need the loop. Will review again in a bit.